### PR TITLE
US975260: Refactor HPE references

### PR DESCRIPTION
**Pull request by raghavendran-gopalakrishnan**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=975260

### CODE REVIEW

**Comment by raghavendran-gopalakrishnan** [comment]
Yet to update the javadocs.

File: pom.xml [hunk 1]
```diff
@@ -211,64 +211,64 @@
                 <version>3.1.8</version>
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi</groupId>
+                <groupId>com.github.cafapi.common.api</groupId>
                 <artifactId>caf-api</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 2]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.cipher</groupId>
+                <groupId>com.github.cafapi.common.ciphers</groupId>
                 <artifactId>cipher-null</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 3]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.codec</groupId>
+                <groupId>com.github.cafapi.common.codecs</groupId>
                 <artifactId>codec-json</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 4]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
-                <artifactId>config-caf</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <groupId>com.github.cafapi.common.config</groupId>
+                <artifactId>config-source</artifactId>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 5]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
+                <groupId>com.github.cafapi.common.config</groupId>
                 <artifactId>config-decoder</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 6]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
+                <groupId>com.github.cafapi.common.config.sources</groupId>
                 <artifactId>config-file</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 7]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
-                <artifactId>config-system</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <groupId>com.github.cafapi.common.bootstrapconfigs</groupId>
+                <artifactId>bootstrapconfig-system</artifactId>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 8]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.decoder</groupId>
+                <groupId>com.github.cafapi.common.decoders</groupId>
                 <artifactId>decoder-js</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 9]
```diff
@@ -211,64 +211,64 @@
...
                 <groupId>com.github.cafapi.logging</groupId>
                 <artifactId>caf-logging-logback</artifactId>
                 <version>2.0.0-238</version>
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.util</groupId>
+                <groupId>com.github.cafapi.common.util</groupId>
                 <artifactId>util-moduleloader</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 10]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.util</groupId>
+                <groupId>com.github.cafapi.common.util</groupId>
                 <artifactId>util-naming</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 11]
```diff
@@ -211,64 +211,64 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.util</groupId>
+                <groupId>com.github.cafapi.common.util</groupId>
                 <artifactId>util-ref</artifactId>
-                <version>3.2.0-SNAPSHOT</version>
+                <version>4.0.0-975260-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

**Comment by raghavendran-gopalakrishnan** [comment]
To do:
Consolidate javadocs of caf-commons and worker-framework.
Reversion to major

File: worker-core/pom.xml [hunk 12]
```diff
@@ -21,14 +21,14 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>com.github.workerframework</groupId>
+    <groupId>com.github.workerframework.worker</groupId>
```
Comment (dermot-hardy) [inline worker-core/pom.xml]: This extra `.worker` is bothersome.  Put these in a spreadsheet and let's go through them.

Comment (raghavgopal) [reply worker-core/pom.xml]: Ok let's review them today.

File: pom.xml [hunk 13]
```diff
@@ -211,115 +211,115 @@
                 <version>3.1.8</version>
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi</groupId>
+                <groupId>com.github.cafapi.common.api</groupId>
```
Comment (dermot-hardy) [inline pom.xml]: This makes me realize we made a mistake in `caf-common`.  We need to discuss this.

File: worker-core/readme.md [hunk 14]
```diff
@@ -83,7 +83,7 @@ the current input queue. Default is True.
  The following command-line should start the application:
  
  ```
- java -cp "*" com.hpe.caf.worker.core.WorkerApplication server [settings.yaml]
+ java -cp "*" WorkerApplication server [settings.yaml]
```
Comment (dermot-hardy) [inline worker-core/readme.md]:
```suggestion
 java -cp "*" com.github.workerframework.core.WorkerApplication server [settings.yaml]
```

File: worker-core/readme.md [hunk 15]
```diff
@@ -707,14 +707,14 @@ different results.
...
+ package com.github.test.worker;
 
  import com.hpe.caf.api.Codec;
  import com.hpe.caf.api.ConfigurationException;
  import com.hpe.caf.api.ConfigurationSource;
  import com.hpe.caf.api.worker.DataSource;
- import com.hpe.caf.api.worker.WorkerException;
- import com.hpe.caf.api.worker.WorkerFactoryProvider;
+ import WorkerException;
+ import WorkerFactoryProvider;
```
Comment (dermot-hardy) [inline worker-core/readme.md]:
```suggestion
 import com.github.workerframework.api.WorkerException;
 import com.github.workerframework.api.WorkerFactoryProvider;
```

File: worker-core/readme.md [hunk 16]
```diff
@@ -755,14 +755,14 @@ different results.
...
- import com.hpe.caf.api.Codec;
- import com.hpe.caf.api.CodecException;
- import com.hpe.caf.api.worker.Worker;
- import com.hpe.caf.api.worker.WorkerException;
- import com.hpe.caf.test.worker.shared.TestWorkerTask;
- import com.hpe.caf.test.worker.shared.TestWorkerResult;
+ import com.github.cafapi.common.api.Codec;
+ import com.github.cafapi.common.api.CodecException;
+ import Worker;
+ import WorkerException;
```
Comment (dermot-hardy) [inline worker-core/readme.md]:
```suggestion
 import com.github.workerframework.api.Worker;
 import com.github.workerframework.api.WorkerException;
```

File: worker-core/readme.md [hunk 17]
```diff
@@ -849,8 +849,8 @@ different results.
  be `WorkerFactoryProvider`, and the file contents will be a single line which
  consists of the fully qualified class name of your implementation. So in this
  case, the file will be called
- `META-INF/services/com.hpe.caf.api.worker.WorkerFactoryProvider` and will
- have the line `com.hpe.caf.test.worker.TestWorkerFactoryProvider`.
+ `META-INF/services/WorkerFactoryProvider` and will
```
Comment (dermot-hardy) [inline worker-core/readme.md]:
```suggestion
 `META-INF/services/com.github.workerframework.api.WorkerFactoryProvider` and will
```

File: pom.xml (hunk 13 above)
Comment (raghavgopal) [reply pom.xml]: Ok it was supposed to be just `com.github.cafapi.common` for all caf-common artifacts.

Comment (dermot-hardy) [reply pom.xml]: Right

Comment (raghavgopal) [reply pom.xml]: Will fix them.

File: pom.xml [hunk 18]
```diff
@@ -211,115 +211,115 @@
                 <version>3.1.8</version>
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>caf-api</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 19]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.cipher</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>cipher-null</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 20]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.codec</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>codec-json</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 21]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
-                <artifactId>config-caf</artifactId>
-                <version>3.2.0-684</version>
+                <groupId>com.github.cafapi.common</groupId>
+                <artifactId>config-source</artifactId>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 22]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>config-decoder</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 23]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>config-file</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 24]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.config</groupId>
-                <artifactId>config-system</artifactId>
-                <version>3.2.0-684</version>
+                <groupId>com.github.cafapi.common</groupId>
+                <artifactId>bootstrapconfig-system</artifactId>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 25]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.decoder</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>decoder-js</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 26]
```diff
@@ -211,115 +211,115 @@
...
                 <groupId>com.github.cafapi.logging</groupId>
                 <artifactId>caf-logging-logback</artifactId>
                 <version>2.0.0-238</version>
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.util</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>util-moduleloader</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 27]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.util</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>util-naming</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 28]
```diff
@@ -211,115 +211,115 @@
...
             </dependency>
             <dependency>
-                <groupId>com.github.cafapi.util</groupId>
+                <groupId>com.github.cafapi.common</groupId>
                 <artifactId>util-ref</artifactId>
-                <version>3.2.0-684</version>
+                <version>4.0.0-975260-1-SNAPSHOT</version>
```
Comment (raghavgopal) [inline pom.xml]:
```suggestion
                <version>4.0.0-SNAPSHOT</version>
```

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**